olive/systems/azureml/aml_system.py

```diff
@@ -657,9 +666,12 @@ def _create_metric_component(
 
         # prepare code
         script_name = "aml_evaluation_runner.py"
+        tmp_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
nit: needless since the pass runner doesn't have this line?

Reply:
This is a different behavior from pass runner because actually this tmp_dir is [`tmp_dir / metric.name`](https://github.com/microsoft/Olive/blob/main/olive/systems/azureml/aml_system.py#L630) but we don't create them until `copy_code` is called.